src/EFCore.PG/Scaffolding/Internal/NpgsqlDatabaseModelFactory.cs

```diff
@@ -543,9 +552,43 @@ ELSE pg_get_expr(indpred, cls.oid)
                             .GetValueOrDefault<uint[]>("indclass")
                             .Select(oid => opClasses.TryGetValue(oid, out var opc) && !opc.IsDefault ? opc.Name : null)
                             .ToArray();
+
                         if (opClassNames.Any(op => op != null))
                             index[NpgsqlAnnotationNames.IndexOperators] = opClassNames;
 
+                        var columnCollations = record
+                            .GetValueOrDefault<uint[]>("indcollation")
+                            .Select(oid => collations.TryGetValue(oid, out var collation) && !string.Equals(collation, "default") ? collation : null)
+                            .ToArray();
+
+                        if (columnCollations.Any(coll => coll != null))
+                            index[NpgsqlAnnotationNames.IndexCollation] = columnCollations;
+
+                        if (record.GetValueOrDefault<bool>("amcanorder"))
+                        {
+                            var options = record.GetValueOrDefault<ushort[]>("indoption");
+
+                            // The first bit specifies whether values are sorted in descending order.
+                            const ushort indoptionDescFlag = 0x0001;
```

Comment:
FWIW I'm not sure it's correct to assume that if `amcanorder` is true, then `indoption` contains bits in this configuration. Another index method with `amcanorder`=true can still have `indoption` with a completely different representation. Since at the moment we only know of one such method (`btree`), the `amcanorder` is effectively a check for `btree` - so it may be a bit safer to explicitly do that instead.

But I don't think it's very important or that we should block the PR for it. Whenever a new index method comes we can take care of it then.

Reply:
I'm pretty sure I've linked this before, but I'm on my phone and couldn't find it. The Postgres code suggests this is always the case when an index supports ordering. See https://github.com/postgres/postgres/blob/aad21d4c3cd84e7c2a28d239048d2e2f494be47e/src/include/catalog/pg_index.h#L70-L74

Also, the JDBC adapter does assume this as well, so I'm pretty sure it's safe. But if you want me to remove it, I can 😊

Reply:
No it's fine :) I was just writing to get it out... I think it's totally fine not to expect EFCore.PG to fully support new index methods out-of-the-box without any modification - when it happens we'll make the proper modifications.